apis/kueue/v1beta1/resourceflavor_types.go

```diff
@@ -0,0 +1,62 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+//+kubebuilder:object:root=true
+//+kubebuilder:resource:scope=Cluster,shortName={flavor,flavors}
+
+// ResourceFlavor is the Schema for the resourceflavors API.
+type ResourceFlavor struct {
+	metav1.TypeMeta   `json:",inline"`
```

Comment:
Can we have a switch to turn off a resource flavor in all of the queues (possibly as a follow up PR)?

Reply:
That's beyond the scope of this PR and it deserves it's own kep with stories.